integration-tests/main/src/main/java/io/quarkus/it/jpa/CustomTypeEntity.java

```diff
@@ -0,0 +1,69 @@
+package io.quarkus.it.jpa;
+
+import java.math.BigInteger;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+import org.hibernate.annotations.TypeDefs;
+
+@Entity
+@TypeDef(typeClass = BigIntType.class, name = "bigIntType1")
+@TypeDefs({
```

Comment:
I think you should use three different user types, not three times the same one, to make sure you really register them all.

Reply:
I can work on that.

Reply:
Done.